src/github.com/mozilla-services/pushgo/simplepush/dynamo_store.go

```diff
@@ -0,0 +1,507 @@
+	tablename     string
+	readProv      int64
+	writeProv     int64
+	closeSignal   chan bool
+	maxChannels   int
+	statusTimeout time.Duration
+	statusIdle    time.Duration
+}
+
+const (
+	RFC_ISO_8601  = "20060102T150405Z"
+	DYNAMO_PREFIX = "DynamoDB_20120810"
+)
+
+func (s *DynamoDBStore) waitUntilStatus(table dynamoTable, status string) (err error) {
+	done := make(chan bool)
+	timeout := time.After(s.statusTimeout)
+	go func() {
+		for {
+			select {
+			case <-done:
+				return
+			default:
+				desc, err := table.DescribeTable()
+				if err != nil {
+					return
+				}
+				if desc.TableStatus == status {
+					done <- true
+					return
+				}
+				time.Sleep(s.statusIdle)
+			}
+		}
+	}()
+	select {
+	case <-done:
+		err = nil
+		break
+	case <-timeout:
+		err = ErrDynamoDBTimeout
+		close(done)
+	}
+	return
+}
+
+func (s *DynamoDBStore) createTable() (err error) {
+	/* There are three primary fields for each row,
+	   uaid = UserAgentID, all registered items must have one of these
+	   chid = Either the ChannelID or " ". In this case " " fills the
+	   		"main" UAID entry, which can carry info like the Proprietary
+	   		Ping data, etc. CHID rows can be accessed by specifing a
+	   		QUERY with chid > " ".
+	   created = UTC().Unix() of the record creation. Used for cleanup.
+	*/
+	tableDesc := dynamodb.TableDescriptionT{
+		TableName: s.tablename,
+		AttributeDefinitions: []dynamodb.AttributeDefinitionT{
+			dynamodb.AttributeDefinitionT{UAID_LABEL, "S"},
+			dynamodb.AttributeDefinitionT{CHID_LABEL, "S"},
+			dynamodb.AttributeDefinitionT{MODD_LABEL, "N"},
+		},
+		KeySchema: []dynamodb.KeySchemaT{
+			dynamodb.KeySchemaT{UAID_LABEL, "HASH"},
+			dynamodb.KeySchemaT{CHID_LABEL, "RANGE"},
+		},
+		GlobalSecondaryIndexes: []dynamodb.GlobalSecondaryIndexT{
+			dynamodb.GlobalSecondaryIndexT{
+				IndexName: "uaid-modified-index",
+				KeySchema: []dynamodb.KeySchemaT{
+					dynamodb.KeySchemaT{UAID_LABEL, "HASH"},
+					dynamodb.KeySchemaT{MODD_LABEL, "RANGE"},
+				},
+				Projection: dynamodb.ProjectionT{"ALL"},
+				ProvisionedThroughput: dynamodb.ProvisionedThroughputT{
+					ReadCapacityUnits:  s.readProv,
+					WriteCapacityUnits: s.writeProv,
+				},
+			},
+		},
+		ProvisionedThroughput: dynamodb.ProvisionedThroughputT{
+			ReadCapacityUnits:  s.readProv,
+			WriteCapacityUnits: s.writeProv,
+		},
+	}
+
+	pk, err := tableDesc.BuildPrimaryKey()
+	if err != nil {
+		s.logger.Panic(DB_LABEL, "Could not create primary key",
+			LogFields{"error": err.Error()})
+		return
+	}
+	s.table = s.server.NewTable(tableDesc.TableName, pk)
+	_, err = s.server.CreateTable(tableDesc)
+	if err != nil {
+		s.logger.Panic(DB_LABEL, "Could not create",
+			LogFields{"error": err.Error()})
+		return
+	}
+
+	err = s.waitUntilStatus(s.table, "ACTIVE")
+	if err != nil {
+		s.logger.Panic("dynamo", "Could not file table.",
+			LogFields{"error": err.Error()})
+		return
+	}
+	return
+}
+
+// NewEmcee creates an unconfigured memcached adapter.
```

Comment:
Well, NewEmcee does it too.

![default](https://cloud.githubusercontent.com/assets/35755/6474759/7e7de900-c1b8-11e4-9a8c-8b1575d5478a.png)